libs/browser-wallet/src/frontend/routes/onboarding/components/mnemonic-page/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { HiddenContainer } from '@/components/hidden-container';
+import { OnboardingPage } from '@/components/pages/onboarding-page';
+
+import { FULL_ROUTES } from '../../../route-names';
+import { SaveMnemonicForm } from './save-mnemonic-form';
+import { WalletCreated } from './wallet-created';
+import { useJsonRpcClient } from '@/contexts/json-rpc/json-rpc-context';
+import { createWallet } from '@/lib/create-wallet';
+
+export const locators = {
+  saveMnemonicDescription: 'save-mnemonic-description',
+};
+
+export const MnemonicPage = ({ mnemonic }: { mnemonic: string | null }) => {
+  const [showSuccess, setShowSuccess] = useState(false);
```

Comment:
non blocking: I'd like to try and avoid writing success/error/loading state  handling over and over again. You could look at using react query here (useMutation)